apps/performance/default-onpush/src/app/name-input.component.ts

```diff
@@ -0,0 +1,43 @@
+import { CDFlashingDirective } from '@angular-challenges/shared/directives';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Output,
+} from '@angular/core';
+import { FormsModule } from '@angular/forms';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatInputModule } from '@angular/material/input';
+@Component({
+  selector: 'app-name-input',
+  standalone: true,
+  imports: [
+    MatFormFieldModule,
+    FormsModule,
+    MatInputModule,
+    CDFlashingDirective,
+  ],
+  template: `
+    <mat-form-field class="w-full" cd-flash>
+      <input
+        placeholder="Add one member to the list"
+        matInput
+        type="text"
+        [(ngModel)]="label"
+        (keydown)="handleKey($event)" />
+    </mat-form-field>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class NameInputComponent {
+  @Output() newNameEvent = new EventEmitter<string>();
+
+  label = '';
+
+  handleKey(event: KeyboardEvent) {
```

Comment:
You can use the `(keydown.enter)` event in your template, so you don't have to check if the pressed key is `Enter`. More information about this [here](https://www.digitalocean.com/community/tutorials/angular-binding-keyup-keydown-events).

Reply:
Refactored, and thanks for that! Something new I learnt today 😄